src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
I don't really like the idea of hardcoding "SR" here, as this tool is more general. If we want SR we should allow for it to be passed in via the message. Which would mean all we need to do is remove the quoting here and enforce that the message also contain quotes if it needs them.